pkg/discovery/agent.go

```diff
@@ -104,8 +101,6 @@ func (a *Agent) CheckHealth() error {
 
 // Run the agent loop
 func (a *Agent) Run() {
-	validator := validateAPI()
-	coreAgent.RegisterAPIValidator(validator)
```

Comment:
Validating APIs like this does not work for our use case because it looks for the externalAPIID on the API Service. When the DA starts it gets all api services found for the environment and caches it. It would be nice if the sdk could be extended to implement use a custom caching policy.

Reply:
Using a separate cache and giving them keys that work for Mulesoft may be what you need.

Reply:
Thanks Jason. Are we missing out on anything useful by not using the validator in the sdk? It looks like it takes care of removing apis that are not found as well, which we do not have support for.

Reply:
It is just for cleaning up, but that is for removing them on the Central side, when they are no longer on the remote gateway.

Reply:
Also, with the external ID issue, you can look into utilizing the Primary Key in lieu of the API ID.